File: Study/Programming Skills/11.Roman to Integer/Go/main.go
func romanToInt(s string) int {
  roman := map[byte]int{
    'I': 1,
    'V': 5,
    'X': 10,
    'L': 50,
    'C': 100,
    'D': 500,
    'M': 1000,
  }
  n := len(s)
  if n == 0 {
    return 0
  }
  res := roman[s[n-1]]
  for i := n - 2; i >= 0; i-- {
    if roman[s[i]] < roman[s[i+1]] {
      res -= roman[s[i]]
    }
    if roman[s[i]] >= roman[s[i+1]] {
      res += roman[s[i]]
    }
  }
  return res
    
}

File: Study/Programming Skills/11.Roman to Integer/Rust/main.rs
impl Solution {
    pub fn roman_to_int(s: String) -> i32 {
        let roman = vec![
            ('I', 1), ('V', 5), ('X', 10), ('L', 50), ('C', 100), ('D', 500), ('M', 1000)
        ].into_iter().collect::<std::collections::HashMap<_, _>>();
        
        let mut result = 0;
        let mut prev_value = 0;

        for c in s.chars() {
            let value = *roman.get(&c).unwrap();
            if value > prev_value {
                result += value - 2 * prev_value;
            } else {
                result += value;
            }
            prev_value = value;
        }

        result
    }
}
